pkg/etcdcli/cached_members_test.go

```diff
@@ -0,0 +1,314 @@
+package etcdcli
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	u "github.com/openshift/cluster-etcd-operator/pkg/testutils"
+	"github.com/stretchr/testify/require"
+
+	"go.etcd.io/etcd/api/v3/etcdserverpb"
+)
+
```

Comment:
you're still missing a test that tests the thing that's actually needed to be tested in the cache: the refresh mechanism :)

suggestion: setup with some members, assert it works, sleep (maybe find a more resilient way), change the members in the client, assert the result has changed.

Reply:
done 👍🏽